### ISSUE TITLE
Overrides not working as intended

### ISSUE THREAD

**Issue by tHBp** [eslint/eslint#11934]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.2.0
* **npm Version:** 6.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "rules": {
        "quote-props": ["error", "as-needed"]
    },
    "overrides": [{
        "files": ["**/someDirectoryName/**"],
        "rules": {
            "quote-props": 0
        }
    }]
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// someDirectoryName/dummy-file.js
var a = {
  "dummyString": "Dummy String value"
}
```
Command:

```bash
eslint -c .eslintrc.json someDirectoryName/
```
Overrides are not applied. Tried ``` "files": ["**/someDirectoryName/dummy-file.js"]``` and ``` "files": ["someDirectoryName/dummy-file.js"]``` too. Both didn't work.

However, it works if I specify the file name only, like below:
```js
"overrides": [{
        "files": ["dummy-file.js"]
```

Let me know if more information is needed.

**Comment by mysticatea** [comment]
Thank you for your report.

However, I couldn't reproduce it.

```
~\dev\sandbox> cat .\.eslintrc.js
module.exports = {
    "root": true,
    "rules": {
        "quote-props": ["error", "as-needed"]
    },
    "overrides": [{
        "files": ["**/someDirectoryName/**"],
        "rules": {
            "quote-props": 1
        }
    }]
}
~\dev\sandbox> cat .\someDirectoryName\dummy-file.js
var a = {
    "dummyString": "Dummy String value"
  }
~\dev\sandbox> eslint .\someDirectoryName\

C:\Users\t-nagashima.AD\dev\sandbox\someDirectoryName\dummy-file.js
  2:5  warning  Unnecessarily quoted property 'dummyString' found  quote-props

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

~\dev\sandbox>
```

(I just tweaked the `overrides` setting as a warning to clarify the result. The result is a warning instead of an error, so the `overrides` setting was applied.)

Would you check your case again or would you provide a repro repository?

**Comment by tHBp** [comment]
Hi @mysticatea 

Thanks for quick revert.

Can you try putting configuration file in some other directory as below:

```bash
eslint -c someOtherDirectory/.eslintrc.json someDirectoryName/
```

**Comment by mysticatea** [comment]
Ah, I got it.

So this is a duplicate of https://github.com/eslint/eslint/issues/11558.
File paths in config files are relative from the config file.

**Comment by tHBp** [comment]
@mysticatea I couldn't find the above marked issue in the first go. My bad. Sorry for creating a separate issue for this.

If this is the intended behavior, how can I go about fixing this issue since passing ```"../"``` in override pattern breaks things by throwing the following error: ```Error: Invalid override pattern (expected relative path not containing '..')```

**Comment by tHBp** [comment]
Also, this behavior is erratic at best. I have a couple of custom rules for which overrides seem to work as expected.
